### PR TITLE
add function to get all label names in db

### LINKED ISSUE #279: Add a method for getting all label names
This has to be implemented in the Querier, like https://github.com/prometheus/tsdb/blob/master/querier.go#L34-L35.  
It's needed to implement https://github.com/prometheus/prometheus/issues/2038.

### PR DESCRIPTION
**Pull request by sipian**
Fixes #279 

This PR fixes for the test cases that were failing in https://github.com/prometheus/tsdb/pull/296

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
@sipian would you mind rebasing and I will review. if you don't have time right now let me know and I will take over. Thanks

in the description it would be better to use something like "fixes #279"

https://help.github.com/articles/closing-issues-using-keywords/

**Comment by sipian** [comment]
@krasi-georgiev I have rebased commits to add DCO. 
And, yeah will follow the keywords in description.

File: db.go [hunk 1]
```diff
@@ -469,6 +469,14 @@ func (db *DB) getBlock(id ulid.ULID) (*Block, bool) {
 	return nil, false
 }
 
+func (db *DB) getLabelNames() []string {
```
Comment (krasi-georgiev) [inline db.go]: I don't see where are we using this?

File: block.go [hunk 2]
```diff
@@ -88,6 +88,12 @@ type IndexReader interface {
 	Close() error
 }
 
+type IndexExtra interface {
```
Comment (krasi-georgiev) [inline block.go]: maybe `IndexLabelReader`

File: index/index.go [hunk 3]
```diff
@@ -875,6 +875,18 @@ func (r *Reader) LabelValues(names ...string) (StringTuples, error) {
 	return st, nil
 }
 
+func (r *Reader) LabelNames() []string {
+	const sep = "\xff"
```
Comment (krasi-georgiev) [inline index/index.go]: this cons is used in many other places , maybe a good follow up PR would be to make ti global.
It is already in the labels package to maybe can just export is and use it everywhere  as labels.Separator

**Comment by sipian** [comment]
@krasi-georgiev 
I'm a bit busy right now with the e2e tool. 
If this PR is not urgent, I would like to get back to this later.

**Comment by krasi-georgiev** [comment]
superseded by https://github.com/prometheus/tsdb/pull/369

**Pull request closed without merging**